### PR TITLE
Spike trying to make scalatest suite type agnostic

### PR DESCRIPTION
**Pull request by chbatey**
Spiked trying to create this base class work with any of the SpecLike traits from ScalaTest

Most of the changes are to the tests that use it, will point out the buts to look at

### CODE REVIEW

File: akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitScalaTestSpec.scala [hunk 1]
```diff
@@ -20,13 +18,9 @@ import org.scalatest.time.Span
...
  * to your project to use this.
- *
- * This is a `WordSpec`, other ScalaTest styles can be implemented in a similar abstract class like this one.
- * For example `with FlatSpecLike with BeforeAndAfterAll` and implementing the `afterAll` method by calling
- * `testKit.shutdownTestKit()`.
  */
-abstract class ActorTestKitWordSpec(testKit: ActorTestKit) extends ActorTestKitBase(testKit)
-  with WordSpecLike with Matchers with BeforeAndAfterAll with ScalaFutures with Eventually {
+abstract class ActorTestKitScalaTestSpec(testKit: ActorTestKit) extends ActorTestKitBase(testKit)
+  with TestSuite with Matchers with BeforeAndAfterAll with ScalaFutures with Eventually {
```
Comment (chbatey) [inline akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitScalaTestSpec.scala]: mixing in TestSuite fixes the compile looking for run method

File: akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitScalaTestSpec.scala [hunk 2]
```diff
@@ -52,5 +46,11 @@ abstract class ActorTestKitWordSpec(testKit: ActorTestKit) extends ActorTestKitB
    */
   implicit val patience: PatienceConfig =
     PatienceConfig(testKit.testKitSettings.DefaultTimeout.duration, Span(100, org.scalatest.time.Millis))
+
+  // FIXME document to call super if override
+  override protected def afterAll(): Unit = {
```
Comment (chbatey) [inline akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitScalaTestSpec.scala]: I think this was missing, i didn't see where it was shutdown

File: akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala [hunk 3]
```diff
@@ -13,7 +13,7 @@ import org.scalatest.Matchers
 import org.scalatest.WordSpec
 import org.scalatest.WordSpecLike
 
-class ActorTestKitSpec extends ActorTestKitWordSpec {
+class ActorTestKitSpec extends ActorTestKitScalaTestSpec with WordSpecLike {
```
Comment (chbatey) [inline akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala]: We should probably create a combined ActorTestKitScalaTestSpec with WordSpecLike but users can choose there own

File: akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitScalaTestSpec.scala (hunk 1 above)
Comment (patriknw) [reply akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitScalaTestSpec.scala]: ah, I thought I had tried all things. Probably got lost in the trait jungle. Thanks

File: akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala (hunk 3 above)
Comment (patriknw) [reply akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala]: I think this is good. Are there alternatives to `Matchers` also, or is that always wanted?

File: akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitScalaTestSpec.scala [hunk 4]
```diff
@@ -52,5 +46,11 @@ abstract class ActorTestKitWordSpec(testKit: ActorTestKit) extends ActorTestKitB
    */
   implicit val patience: PatienceConfig =
     PatienceConfig(testKit.testKitSettings.DefaultTimeout.duration, Span(100, org.scalatest.time.Millis))
+
+  // FIXME document to call super if override
```
Comment (patriknw) [inline akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitScalaTestSpec.scala]: yes please

File: akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitScalaTestSpec.scala (hunk 2 above)
Comment (patriknw) [reply akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitScalaTestSpec.scala]: indeed, added it as abstract in ActorTestKitBase but forgot our own impl of it

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala (hunk 3 above)
Comment (chbatey) [reply akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala]: Some prefer regular assertions so perhaps this and ScalaFutures should be in our private impl as they add a lot of implicits

Comment (patriknw) [reply akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala]: it was nice to change the patience in one place, or is that in Eventually and we could keep that?

one of my goals was to not have to much internal infrastructure for our tests but use the testkit as is, so the drawback would be a bit more copy paste of mixins into each test class

**Comment by chbatey** [comment]
@patriknw i've pushed the changes for the other typed modules. Do you want to merge this into your branch?

**Comment by patriknw** [comment]
yes, let's merge those things before master moves more. I can do appropriate squashing in my branch, need rebase anyway

we can decide about the Matchers in a separate PR (I think it's pretty convenient to have the full package, and those that have other needs could create their own base class)

File: akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala (hunk 3 above)
Comment (chbatey) [reply akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala]: We could leave it as is then as it is fairly straight forward for users to create a similar base class for their tests

**Pull request merged**